source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1078,21 +1078,32 @@ void Filter::sendTrailers(ProcessorState& state, const Http::HeaderMap& trailers
   stats_.stream_msgs_sent_.inc();
 }
 
-void Filter::logGrpcStreamInfo() {
+void Filter::logStreamInfoBase(Envoy::StreamInfo::StreamInfo* stream_info) {
+  if (stream_info == nullptr || logging_info_ == nullptr) {
+    return;
+  }
+
+  const auto& upstream_meter = stream_info->getUpstreamBytesMeter();
+  if (upstream_meter != nullptr) {
+    logging_info_->setBytesSent(upstream_meter->wireBytesSent());
+    logging_info_->setBytesReceived(upstream_meter->wireBytesReceived());
+  }
+  // Only set upstream host in logging info once.
+  if (logging_info_->upstreamHost() == nullptr) {
+    logging_info_->setUpstreamHost(stream_info->upstreamInfo()->upstreamHost());
+  }
+}
+
+void Filter::logStreamInfo() {
   if (!config().grpcService().has_value()) {
+    // HTTP service
+    logStreamInfoBase(stream_info_http_);
```

Comment:
Do we need to cache `stream_info_http_` and `setStreamInfo` function?

Can we get access to this through existing members and active_requests_, similar to  gRPC?

Reply:
Sure. We can add a new interface: client->getStreamInfo() to retrieve streamInfo() from active_request_.